lib/galaxy/tool_util/verify/asserts/text.py

```diff
@@ -74,11 +74,11 @@ def assert_has_line(
 
 def assert_has_n_lines(
     output,
-    n: Optional[int] = None,
-    delta: int = 0,
-    min: Optional[int] = None,
-    max: Optional[int] = None,
-    negate: bool = False,
+    n: Optional[str] = None,
```

Comment:
Only when coming from the XML right? YAML tools, workflow tests, etc... will be properly typed I imagine. I would use Union types to capture this if it is important to lose the typing for XML.

Reply:
Yeah - this feels like you're throwing out a lot of good stuff to support a kind of edge case. Can you just update the size parameter to have a union type and make sure the annotation check is skipping complex types?

Reply:
Will try.

Reply:
Have added bool/int to the parameters in addition to str. 

I'm not sure about the code that I removed from the linting code. Since this is only for xml I'm unsure if its really needed. Feels a bit redundant with the xsd linter. Also for xml we will always have str .. so I don't see this failing in any case. 

What do you think?